File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.py
def solution(n):
    answer = 0
    s = [s_ for s_ in bin(n)[2:] if s_=='1']
    cnt = 0
    idx = 1
    print(len(s))
    while True:
        n += idx
        cnt = len([s_ for s_ in bin(n)[2:] if s_=='1'])
        if cnt == len(s):
            answer = n
            break
    return answer

File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.java
class Solution {


    public static int getOnes(int n){
        int numberOfOnes = 0;
        while (n != 0){
            if(n%2==1){
                numberOfOnes+=1;
            }
            n /= 2;
        }
        return numberOfOnes;
    }

    public int solution(int n) {
        int answer = 0;
        int target = getOnes(n);
        int next = n;
        while (true){
            next+=1;
            int compare = getOnes(next);
            if(compare==target){
                answer = next;
                break;
            }
        }

        return answer;
    }
}
